traverse_test.go

```diff
@@ -46,3 +47,262 @@ func TestNetworksWithInvalidSearchTree(t *testing.T) {
+		}
+
+		expectedIPs := []string{
+			"1.1.1.1/32",
+			"1.1.1.2/31",
+			"1.1.1.4/30",
+			"1.1.1.8/29",
+			"1.1.1.16/28",
+			"1.1.1.32/32",
+		}
+
+		assert.Equal(t, expectedIPs, innerIPs)
+		assert.Nil(t, n.Err())
+	}
+}
+
+func TestNetworksWithinSlash32V4SearchInV4Db(t *testing.T) {
+	_, network, err := net.ParseCIDR("1.1.1.1/32")
+	assert.Nil(t, err)
+
+	for _, recordSize := range []uint{24, 28, 32} {
+		fileName := testFile(fmt.Sprintf("MaxMind-DB-test-ipv4-%d.mmdb", recordSize))
+		reader, err := Open(fileName)
+		require.Nil(t, err, "unexpected error while opening database: %v", err)
+		defer reader.Close()
+
+		n := reader.NetworksWithin(network)
+		var innerIPs []string
+
+		for n.Next() {
+			record := struct {
+				IP string `maxminddb:"ip"`
+			}{}
+			network, err := n.Network(&record)
+			assert.Nil(t, err)
+			assert.Equal(t, record.IP, network.IP.String(),
+				"expected %s got %s", record.IP, network.IP.String(),
+			)
+			innerIPs = append(innerIPs, network.String())
+		}
+
+		expectedIPs := []string{
+			"1.1.1.1/32",
+		}
+
+		assert.Equal(t, expectedIPs, innerIPs)
+		assert.Nil(t, n.Err())
+	}
+}
+
+func TestNetworksWithinSlash32V4SearchInV6Db(t *testing.T) {
+	_, network, err := net.ParseCIDR("1.1.1.1/32")
+	assert.Nil(t, err)
+
+	for _, recordSize := range []uint{24, 28, 32} {
+		fileName := testFile(fmt.Sprintf("MaxMind-DB-test-mixed-%d.mmdb", recordSize))
+		reader, err := Open(fileName)
+		require.Nil(t, err, "unexpected error while opening database: %v", err)
+		defer reader.Close()
+
+		n := reader.NetworksWithin(network)
+		var innerIPs []string
+
+		for n.Next() {
+			record := struct {
+				IP string `maxminddb:"ip"`
+			}{}
+			network, err := n.Network(&record)
+			assert.Nil(t, err)
+			innerIPs = append(innerIPs, network.String())
+		}
+
+		expectedIPs := []string{
+			"1.1.1.1/32",
+		}
+
+		assert.Equal(t, expectedIPs, innerIPs)
+		assert.Nil(t, n.Err())
+	}
+
+}
+func TestNetworksWithinSlash128V6SearchInV6Db(t *testing.T) {
+	_, network, err := net.ParseCIDR("::1:ffff:ffff/128")
+	assert.Nil(t, err)
+
+	for _, recordSize := range []uint{24, 28, 32} {
+		fileName := testFile(fmt.Sprintf("MaxMind-DB-test-ipv6-%d.mmdb", recordSize))
+		reader, err := Open(fileName)
+		require.Nil(t, err, "unexpected error while opening database: %v", err)
+		defer reader.Close()
+
+		n := reader.NetworksWithin(network)
+		var innerIPs []string
+
+		for n.Next() {
+			record := struct {
+				IP string `maxminddb:"ip"`
+			}{}
+			network, err := n.Network(&record)
+			assert.Nil(t, err)
+			assert.Equal(t, record.IP, network.IP.String(),
+				"expected %s got %s", record.IP, network.IP.String(),
+			)
+			innerIPs = append(innerIPs, network.String())
+		}
+
+		expectedIPs := []string{
+			"::1:ffff:ffff/128",
+		}
+
+		assert.Equal(t, expectedIPs, innerIPs)
+		assert.Nil(t, n.Err())
+	}
+}
+
+func TestNetworksWithinV6SearchInV6Db(t *testing.T) {
+	var network = &net.IPNet{IP: make(net.IP, 16), Mask: net.CIDRMask(0, 128)}
```

Comment:
The next three test functions largely duplicate the `Networks()` test. I think we would get better coverage of the expected behaviors by searching within limited subnets for them.